stdlib/join/join_test.flux

```diff
@@ -282,3 +282,103 @@ testcase right_outer_join {
 
     testing.diff(want: want, got: got)
 }
+
+testcase exclusive_group_keys1 {
+    tbl1 =
+        array.from(rows: [{_time: 2022-06-01T00:00:00Z, _value: 1, label: "a", key: 1}])
+            |> group(columns: ["key"])
+    tbl2 =
+        array.from(rows: [{_time: 2022-06-01T00:00:00Z, _value: 0.1, id: "a", key: 2}])
+            |> group(columns: ["key"])
+
+    wantOutput =
+        "
+#datatype,string,long,string,long,double,dateTime:RFC3339,long
+#group,false,false,false,false,false,false,false
+#default,_result,,,,,,
+,result,table,label,v_left,v_right,_time,key
+,,0,a,1,,2022-06-01T00:00:00Z,1
+,,0,a,,0.1,2022-06-01T00:00:00Z,2
+"
+    want = csv.from(csv: wantOutput) |> group(columns: ["key"])
+
+    got =
+        join.tables(
+            method: "full",
+            left: tbl1,
+            right: tbl2,
+            on: (l, r) => l.label == r.id and r._time == l._time,
+            as: (l, r) => {
+                time = if exists l._time then l._time else r._time
+                label = if exists l.label then l.label else r.id
+
+                return {
+                    _time: time,
+                    label: label,
+                    v_left: l._value,
+                    v_right: r._value,
+                    key: r.key,
+                }
+            },
+        )
+
+    testing.diff(want: want, got: got)
+}
+
+testcase exclusive_group_keys2 {
+    tbl1 =
+        array.from(
+            rows: [
+                {
+                    _time: 2022-06-01T00:00:00Z,
+                    _value: 1,
+                    label: "a",
+                    key: 1,
+                    group: "one",
+                },
+            ],
+        )
+            |> group(columns: ["key", "group"])
+    tbl2 =
+        array.from(rows: [{_time: 2022-06-01T00:00:00Z, _value: 0.1, id: "a", key: 2}])
+            |> group(columns: ["key"])
+
+    wantOutput =
+        "
+#datatype,string,long,string,long,double,dateTime:RFC3339,long,string
+#group,false,false,false,false,false,false,true,true
+#default,_result,,,,,,,
+,result,table,label,v_left,v_right,_time,key,group
+,,0,a,1,,2022-06-01T00:00:00Z,1,one
+
+#datatype,string,long,string,long,double,dateTime:RFC3339,long,string
+#group,false,false,false,false,false,false,true,false
+#default,_result,,,,,,,
+,result,table,label,v_left,v_right,_time,key,group
+,,0,a,,0.1,2022-06-01T00:00:00Z,2,
```

Comment:
Should `group` be in the group key here even though there is only nulls?

Reply:
That's a good question. Each output table currently has the same group key as its parent tables. I'm honestly not sure what the right answer is in this weird case.

Reply:
Group key gets set [here](https://github.com/influxdata/flux/blob/4fb8524daf39bb21b80e3ec6e0a686631021e070/stdlib/join/join_fn.go#L156) for every output table, and gets set [here](https://github.com/influxdata/flux/blob/4fb8524daf39bb21b80e3ec6e0a686631021e070/stdlib/join/join_fn.go#L100) or [here](https://github.com/influxdata/flux/blob/4fb8524daf39bb21b80e3ec6e0a686631021e070/stdlib/join/join_fn.go#L111) if one side of a join key is empty.